### ISSUE TITLE
SNOW-636725: panic: crypto/cipher: input not full blocks

### ISSUE THREAD

**Issue by Thomasdezeeuw** [snowflakedb/gosnowflake#632]
### Issue description

See backtrace below.

### Example code

Not available.

### Error log

Trace:
```
panic: crypto/cipher: input not full blocks

goroutine 99 [running]:
crypto/cipher.(*cbcDecrypter).CryptBlocks(0x140003b61a0?, {0x14004800000?, 0x0?, 0x26?}, {0x14004800000?, 0x21?, 0x1400060c978?})
        /opt/homebrew/Cellar/go/1.18.4/libexec/src/crypto/cipher/cbc.go:119 +0x3cc
github.com/snowflakedb/gosnowflake.decryptFile(0x140005a49f0, 0x14000496300?, {0x1400004c6e0, 0x4c}, 0x0, {0x140003b4e00, 0x3b})
        /Users/thomas/go/pkg/mod/github.com/snowflakedb/gosnowflake@v1.6.11/encrypt_util.go:235 +0x408
github.com/snowflakedb/gosnowflake.(*remoteStorageUtil).downloadOneFile(0x0?, 0x14000496300)
        /Users/thomas/go/pkg/mod/github.com/snowflakedb/gosnowflake@v1.6.11/storage_client.go:213 +0x484
github.com/snowflakedb/gosnowflake.(*snowflakeFileTransferAgent).downloadOneFile(0x140002a08c0, 0x14000496300)
        /Users/thomas/go/pkg/mod/github.com/snowflakedb/gosnowflake@v1.6.11/file_transfer_agent.go:986 +0x17c
github.com/snowflakedb/gosnowflake.(*snowflakeFileTransferAgent).downloadFilesParallel.func1(0x2, 0x140002ddd00?)
        /Users/thomas/go/pkg/mod/github.com/snowflakedb/gosnowflake@v1.6.11/file_transfer_agent.go:886 +0x7c
created by github.com/snowflakedb/gosnowflake.(*snowflakeFileTransferAgent).downloadFilesParallel
        /Users/thomas/go/pkg/mod/github.com/snowflakedb/gosnowflake@v1.6.11/file_transfer_agent.go:884 +0x264
```

### Configuration

*Driver version (or git SHA):* gosnowflake@v1.6.11

*Go version:* go version go1.18.4 darwin/arm64

**Comment by Thomasdezeeuw** [comment]
Any update on this? This is blocking our support for Snowflake.

**Comment by Thomasdezeeuw** [comment]
It's been two months now... we're trying to support Snowflake, but without a response this makes it quite difficult.

**Comment by ColmanHumphrey** [comment]
Hey @SimbaGithub @sfc-gh-hchaturvedi @sfc-gh-jfan @sfc-gh-dszmolka @sfc-gh-mkeller @sfc-gh-jbahk @ sfc-gh-sbommakanty:

We're hoping that Snowflake can be a first-class supported database in our product. So far progress is good, but this issue along with https://github.com/snowflakedb/gosnowflake/issues/630 and https://github.com/snowflakedb/gosnowflake/issues/631 are blocking issues for some key aspects of integration with Snowflake. I've tagged you because you've previously interacted with Thomas's issues.

As you can see Thomas has opened a fair few other issues, but these three are the most vital.

Is there a better process we should go through than creating issues in this repo for getting in touch with you all? If this isn't your preferred workflow, we'd be happy to use another line of contact. Of course you don't owe us a response for these issues, but without one it's hard to reliably estimate when we can have full Snowflake integration. 

Thanks!

**Comment by sfc-gh-dszmolka** [comment]
hello @ColmanHumphrey  thank you for raising awareness for this.

From the process perspective: if issues concerning official Snowflake drivers (like gosnowflake here) are raised with Snowflake Support, then the issues are automatically associated with SLA's according to their priorities at submit time; also tracking, regular updates, and everything which one might expect from an official support procedure. Please find the howto on how to submit a case with Support here: https://community.snowflake.com/s/article/How-To-Submit-a-Support-Case-in-Snowflake-Lodge. This might be the preferred approach for more important or burning issues.

In this particular case; I now reached out to the connector dev team internally, regarding all 3 issues mentioned (630, 631, 632). Thank you for bearing with us while they're debugged ! 

I also see there's no repro code snippet associated with any of these cases. For 630 this indeed doesn't seem necessary, but if we could please get at least a minimal reproduction code snippet which leads to the issue for 631 and 632, that might help the troubleshooting effort tremendously. Thank you in advance for providing them, if that's possible.

**Comment by ColmanHumphrey** [comment]
Hey @sfc-gh-dszmolka,

Regarding support: to be honest my prior tickets have not had a ton of engagement. But just a question on that, I'm a little unsure: are you saying you'd prefer that to issues in this repo? Or both are good? The situation is less clear than normal: we are not trying to get our own Snowflake usage improved, we are helping our customers get the most value from Snowflake through our integration. 

> In this particular case; I now reached out to the connector dev team internally, regarding all 3 issues mentioned (630, 631, 632). Thank you for bearing with us while they're debugged !

Fantastic!

> I also see there's no repro code snippet associated with any of these cases. For 630 this indeed doesn't seem necessary, but if we could please get at least a minimal reproduction code snippet which leads to the issue for 631 and 632, that might help the troubleshooting effort tremendously. Thank you in advance for providing them, if that's possible.

We'll work on those. The full code that produced them is large and complex (and somewhat proprietary), so we'll need to do a bit of work to extract out a simpler repro.

Thanks for reaching out!

**Comment by sfc-gh-dszmolka** [comment]
hi @ColmanHumphrey 

> Regarding support: to be honest my prior tickets have not had a ton of engagement.

I sent you a mail related to this to discuss the details because this is really not something which our customers should experience. I would like to see where we could improve.

> are you saying you'd prefer that to issues in this repo? Or both are good? 

Both are good indeed. I'm positive the issues submitted in this repo are also known to the dev team, but I'll let them comment further on that if they wish.
However, the 'official' Support tickets do have certain metrics and obligations tied to them and maybe you could prefer them if you require a bit more tracking and updates on a certain issue. 

Also thank you for the efforts and help with the repro snippets !

**Comment by sfc-gh-dszmolka** [comment]
Hi,
The driver dev team made a new branch of the driver which is hoped to resolve issues #630, #631, #632.
https://github.com/snowflakedb/gosnowflake/tree/fixCipherPanicIssues

Would it be please possible @Thomasdezeeuw to test it and see if it really does ? Would appreciate any feedbacks.

**Comment by Thomasdezeeuw** [comment]
@sfc-gh-dszmolka I think this issue is also fixed, however not 100% sure (I lost the reproduction code). In any case thanks for the work @sfc-gh-ext-simba-lb!

**Comment by sfc-gh-dszmolka** [comment]
hi @Thomasdezeeuw thank you so much for testing the fix for this and the other 2 as well! We can still revisit the issue if you perhaps see it resurface.

**Comment by iamontheinet** [comment]
Has this been resolved? If so, can we please close this issue? Thanks!

**Comment by sfc-gh-dszmolka** [comment]
sadly i have no permissions to close this issue.

**Comment by sfc-gh-dszmolka** [comment]
fixed by #687 earlier
